docs/src/content/en/reference/templates.mdx

```diff
@@ -0,0 +1,222 @@
+
+### Installation
+
+Install a template using the `create-mastra` command:
+
+```bash copy
+npx create-mastra@latest --template template-name
+```
+
+This creates a complete project with all necessary code and configuration.
+
+### Setup Process
+
+After installation:
+
+1. **Navigate to project directory**:
+
+   ```bash copy
+   cd your-project-name
+   ```
+
+2. **Configure environment variables**:
+
+   ```bash copy
+   cp .env.example .env
+   ```
+
+   Edit `.env` with required API keys as documented in the template's README.
+
+3. **Install dependencies** (if not done automatically):
+
+   ```bash copy
+   npm install
+   ```
+
+4. **Start development server**:
+
+   ```bash copy
+   npm run dev
+   ```
+
+### Template Structure
+
+All templates follow this standardized structure:
+
+<FileTree>
+  <FileTree.Folder name="template-name" defaultOpen>
+    <FileTree.Folder name="src" defaultOpen>
+      <FileTree.Folder name="mastra" defaultOpen>
+        <FileTree.Folder name="agents">
+          <FileTree.File name="example-agent.ts" />
+        </FileTree.Folder>
+        <FileTree.Folder name="tools">
+          <FileTree.File name="custom-tool.ts" />
+        </FileTree.Folder>
+        <FileTree.Folder name="workflows">
+          <FileTree.File name="example-workflow.ts" />
+        </FileTree.Folder>
+        <FileTree.File name="index.ts" />
+      </FileTree.Folder>
+    </FileTree.Folder>
+    <FileTree.File name=".env.example" />
+    <FileTree.File name="package.json" />
+    <FileTree.File name="README.md" />
+    <FileTree.File name="tsconfig.json" />
+  </FileTree.Folder>
+</FileTree>
+
+## Creating Templates
+
+### Requirements
+
+Templates must meet these technical requirements:
+
+#### Project Structure
+
+- **Mastra code location**: All Mastra code must be in `src/mastra/` directory
+- **Component organization**:
+  - Agents: `src/mastra/agents/`
+  - Tools: `src/mastra/tools/`
+  - Workflows: `src/mastra/workflows/`
+  - Main config: `src/mastra/index.ts`
+
+#### TypeScript Configuration
+
+Use the standard Mastra TypeScript configuration:
+
+```json filename="tsconfig.json"
+{
+  "compilerOptions": {
+    "target": "ES2022",
+    "module": "ES2022",
+    "moduleResolution": "bundler",
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "skipLibCheck": true,
+    "noEmit": true,
+    "outDir": "dist"
+  },
+  "include": ["src/**/*"]
+}
+```
+
+#### Environment Configuration
+
+Include a `.env.example` file with all required environment variables:
+
+```bash filename=".env.example"
+# OpenAI API key for LLM operations
+OPENAI_API_KEY=your_openai_api_key_here
+
+# Other service API keys as needed
+OTHER_SERVICE_API_KEY=your_api_key_here
+```
+
+### Code Standards
+
+#### LLM Provider
+
+Use OpenAI as the default provider unless demonstrating specific integrations:
+
+```typescript filename="src/mastra/agents/example-agent.ts"
+import { Agent } from '@mastra/core';
```

Comment:
Is this import correct? Should be: `import { Agent } from '@mastra/core/agent';`